File: C#/0367. Valid Perfect Square.cs
public class Solution {
    public bool IsPerfectSquare(int num) {
        if(num<0){
            return false;
        }
        bool flag=false;
        int left=1;
        int right=num;
        double tmp=Math.Sqrt(int.MaxValue);
        while(left<=right){
            int mid=left+(right-left)/2;
            if (mid>tmp){
                right=mid-1;
            }
            else{
                if(mid*mid==num){
                    return true;
                }
                else if(mid*mid>num){
                    right=mid-1;
                }
                else{
                    left=mid+1;
                }
            }
            
        }
        return flag;
    }
}

File: 20190519python/0367. Valid Perfect Square.py
class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num<0:
            return False
        lo=0
        hi=num
        while lo<=hi:
            mid=(lo+hi)//2
            if mid**2==num:
                return True
            elif mid**2<num:
                lo=mid+1
            else:
                hi=mid-1
        return False
